src/App.tsx

```diff
@@ -15,36 +16,110 @@ export const goodsFromServer = [
+}
+
+type SortConfig = {
+  type: string | null;
+  reversed: boolean;
+};
+
 export const App: React.FC = () => {
+  const [sortConfig, setSortConfig] = useState<SortConfig>({
+    type: '',
+    reversed: false,
+  });
+
+  const handleSort = (goods: string[], config: SortConfig) => {
+    const newSortedGoods = [...goods];
+
+    switch (config.type) {
+      case SortType.byAlphabet:
+        newSortedGoods.sort((a, b) => a.localeCompare(b));
+        break;
+      case SortType.byLength:
+        newSortedGoods.sort((a, b) => a.length - b.length);
+        break;
+      default:
+        break;
+    }
+
+    if (config.reversed) {
+      newSortedGoods.reverse();
+    }
+
+    return newSortedGoods;
+  };
+
+  const sortedGoods = handleSort(goodsFromServer, sortConfig);
+
+  const handleReverse = () => {
+    setSortConfig(prevConfig => ({
+      ...prevConfig,
+      reversed: !prevConfig.reversed,
+    }));
+  };
+
+  const handleReset = () => {
+    setSortConfig({ type: '', reversed: false });
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortConfig.type !== 'byAlphabet',
+          })}
+          onClick={() =>
+            setSortConfig({ ...sortConfig, type: SortType.byAlphabet })
+          }
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortConfig.type !== 'byLength',
+          })}
+          onClick={() =>
+            setSortConfig({ ...sortConfig, type: SortType.byLength })
+          }
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !sortConfig.reversed,
+          })}
+          onClick={handleReverse}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortConfig.type || sortConfig.reversed) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={handleReset}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {sortedGoods.map(item => (
+          <li data-cy="Good" key={item}>
```

Comment:
When rendering a list, make sure to provide a unique `key` prop to each element. Using the item itself as a key is not recommended if the items are not guaranteed to be unique. Consider using an index or a unique identifier if available.